fc/mef.py

```diff
@@ -539,54 +261,67 @@ def sc_fun(p,x):
         return np.exp(p[0] * x + p[1])
     
     # Fit parameters
-    err_par = lambda p: err_fun(p, peaks_ch, peaks_mef)
+    err_par = lambda p: err_fun(p, fl_channel, fl_mef)
     res = minimize(err_par, params, options = {'gtol': 1e-6})
 
     # Separate parameters
-    sc_params = res.x
+    beads_params = res.x
 
     # Beads model function
-    sc_beads = lambda x: fit_fun(sc_params, x)
+    beads_model = lambda x: fit_fun(beads_params, x)
 
     # Standard curve function
-    sc = lambda x: sc_fun(sc_params, x)
+    std_crv = lambda x: sc_fun(beads_params, x)
     
-    return (sc, sc_beads, sc_params)
-
-def plot_standard_curve(peaks_ch, 
-                        peaks_mef,
-                        sc_beads,
-                        sc_abs,
-                        xlim = (0., 1023.),
-                        ylim = (1, 1e8)):
+    return (std_crv, beads_model, beads_params)
```

Comment:
> I don't think any of the three first points would make any difference. This is a relatively easy and smooth optimization problem.

It has not been demonstrated to me that these don't make a difference. Especially in light of (1) Prabha's data set where her MEF values change based on the inclusion or exclusion of the 0 peak and (2) the suspicious drop off of fit bead autofluorescence around a gain of 400, and (3) an investigation I did into the different optimization algorithms a while back (see https://wiki.rice.edu/confluence/display/~jts6/JS20150713 and https://github.com/JS3xton/fc/issues/5). I don't know if point 1 and point 2 would fix these things, but I do believe they would make for a more consistent, robust initialization which would be less sensitive to any single peak value. It's possible that the convergence tolerance is low enough that the resulting fit won't change much. I believe these initializations are generally better practice, though, and I am willing to go implement them myself if necessary if there is no good reason why these are worse initializations.

I also don't see a reason not to use norm. Euclidean distance is typically used as an error function for these kinds of optimizations, and anything straying from this should be justified (which the current error function is not). The only argument in favor of the current error function that I see is that it's already done. Again, I'm willing to change it if that is the only objection.

> Also, the first peak seen by this function is not autofluorescence if the user decided to discard the first peak.

I almost mentioned this explicitly. Very true. I still think initializing with a non-zero peak (which should at least result in a `bead_auto` term of comparable units/magnitude to the "true" bead autofluorescence) is much more relevant than arbitrarily choosing 100. I think it would actually be interesting to see if this has any affect on the drop in bead autofluorescence at gains <= 400, as this will force the fitting algorithm to push `bead_auto` down from a non-zero peak, instead of almost certainly pushing it *up* from a value of 100 (which is an order of magnitude low for FL1).

> I don't completely understand the last point. Was this a problem that you actually saw? I don't think this is being a problem currently, otherwise running the example excel file would give the same standard curve three times.

It was a problem for me a long time ago when I tried to generated multiple standards curves in a loop. I just tried it again, and it doesn't seem to be a problem now (three different functions give rise to 3 different transformed values, as expected):

```python
>>> import FlowCal as fc
>>> import numpy as np
>>> x = np.array([250, 400, 600, 800, 950])
>>> y1 = np.exp(0.00925*x + 2) - 1000
>>> y2 = np.exp(0.00925*x + 4) - 1000
>>> y3 = np.exp(0.00925*x + 6) - 1000
>>> y1
array([  -925.37317702,   -701.13259903,    900.74273134,  11088.38073022,
        47411.85496472])
>>> y2
array([   -448.57821849,    1208.34799189,   13044.6946715 ,
         88321.72336081,  356717.91218758])
>>> y3
array([    3074.48647773,    15317.60719802,   102777.03682009,
         659003.22476616,  2642197.72074637])
>>> sc1, bm1, p1 = fc.mef.fit_beads_autofluorescence(x, y1)
>>> sc2, bm2, p2 = fc.mef.fit_beads_autofluorescence(x, y2)
>>> sc3, bm3, p3 = fc.mef.fit_beads_autofluorescence(x, y3)
>>> p1
array([  9.68649682e-03,   1.56445560e+00,   1.00000000e+02])
>>> p2
array([  9.30639475e-03,   3.94362557e+00,   1.00000000e+02])
>>> p3
array([  9.24987943e-03,   6.00008724e+00,   1.00021559e+03])
>>> sc1(500)
606.49956529292444
>>> sc2(500)
5414.4303158582998
>>> sc3(500)
41151.964740359654
>>> fxn = []
>>> for y in (y1, y2, y3):
...  sc, bm, p = fc.mef.fit_beads_autofluorescence(x,y)
...  fxn.append(sc)
...
>>> fxn[0](500)
606.49956529292444
>>> fxn[1](500)
5414.4303158582998
>>> fxn[2](500)
41151.964740359654
```

I still don't understand why it isn't a problem (again, I'm assuming the 2 layers of functions somehow uses the lambda functions to bind `beads_param` at definition time, not run time), but it appears to be OK as far as I can tell, so I don't think we need to worry about it.

Another related thing that I forgot to mention in the last comment: based on investigation here: [https://wiki.rice.edu/confluence/display/~jts6/JS20150713 and https://github.com/JS3xton/fc/issues/5], I think we should use Nelder-Mead as the fitting algorithm instead of the default BGFS.